src/fides/api/migrations/hash_migration_mixin.py

```diff
@@ -0,0 +1,82 @@
+from abc import abstractmethod
+from typing import Any, List, Optional, Tuple
+
+from sqlalchemy import Boolean, Column, Index
+from sqlalchemy.ext.declarative import declared_attr
+
+from fides.api.migrations.hash_migration_tracker import HashMigrationTracker
+from fides.api.schemas.redis_cache import MultiValue
+
+
+class HashMigrationMixin:
+    """
+    A mixin that includes common logic for models that have hashed fields that need be be migrated from using bcrypt to SHA-256.
+    """
+
+    # set existing rows to false but set to true on create since we'll
+    # be using a new hashing function moving forward
+    is_hash_migrated = Column(Boolean, nullable=False, server_default="f", default=True)
+
+    @declared_attr
+    def __table_args__(cls: Any) -> Tuple:
+        """
+        Creates a dynamically named partial index on the table to track unmigrated rows.
+        The size of the index will shrink as more rows are migrated.
+        """
+        return (
+            Index(
+                f"idx_{cls.__tablename__}_unmigrated",  # type: ignore[attr-defined]
+                "is_hash_migrated",
+                postgresql_where=cls.is_hash_migrated.is_(False),  # type: ignore[attr-defined]
+            ),
+        )
```

Comment:
this seems fancy, and i'm not entirely sure how it works, so i'll defer to you on it if you're sure this is the right approach.

generally though, i just wanna double check that creating an index for this column is _worth_ it. i understand that it'll help improve the performance/efficiency of the migration task, but at what cost? will it slow down writes to this table more generally, moving forward?

maybe that negative impact is minimal, maybe i'm overthinking it or thinking about it incorrectly. if you've considered and investigated this, then i'll trust your judgement on it - i just wanted to flag it in case it hadn't been considered yet!